### DISCUSSION TITLE
I don't understand what's wrong with my code

### DISCUSSION THREAD

**Discussion by 6565man** [micropython/micropython#10698] [RP2040 / Pico]
I've tested this script, and it works on the first servo, but when I add more, or use one that isn't the first, it stops working, is this something I'm being an idiot about? Or is there an actual problem?

this is my code:

    from machine import Pin, PWM, ADC
    import utime

    pot1 = ADC(Pin(28))
    pot2 = ADC(Pin(27))
    pot3 = ADC(Pin(26))
    pot4 = ADC(Pin(22))

    servo1 = PWM(Pin(1))
    servo2 = PWM(Pin(2))
    servo3 = PWM(Pin(3))
    servo4 = PWM(Pin(4))

    servo1.freq(50)
    servo2.freq(50)
    servo3.freq(50)
    servo4.freq(50)

    # A .duty_u16 value of c.1350 is zero degrees; 8200 is 180 degrees.
    # This needs the calculation here for clarity! :)

    while True:
        value1 = int(1350 + (pot1.read_u16()/9.57))
        servo1.duty_u16(value1)
        utime.sleep(0.02)

        value2 = int(1350 + (pot2.read_u16()/9.57))
        servo2.duty_u16(value2)
        utime.sleep(0.02)

        value3 = int(1350 + (pot3.read_u16()/9.57))
        servo3.duty_u16(value3)
        utime.sleep(0.02)
    
        value4 = int(1350 + (pot4.read_u16()/9.57))
        servo4.duty_u16(value4)
        utime.sleep(0.02)

**Comment by th3w** [comment]
Hmm, pot4=ADC(Pin(22) ) shouldn‘t work because the pico has only gpio pin 26-28 as usable adc inputs.

**Reply by peterhinch** [reply]
Also I think there's an issue with the PWM pins. Pins 2 and 3 are both assigned to PWM 1.

**Reply by 6565man** [reply]
Thanks

**Reply by 6565man** [reply]
Also though, to your point @th3w only the first one is working, but when I turned it on this morning, without change, the second worked as well.

**Reply by th3w** [reply]
Have you alreade tested with a print out of the values, whether each ADC Channel returns values in the expected range?
So in the While loop replace the ```servox.duty_u16()``` with a ```print(f "Pot x : {valuex}")``` etc.  (replace **x** with the the assigned numbers in you code) and observe the output on the REPL. Maybe increase the sleep time  to 1 seconed or more too, to better detect value changes when manupulating the connected potentiometers.
